src/pages/discord/link/components/view/index.tsx

```diff
@@ -0,0 +1,89 @@
+import { useWallet } from "@solana/wallet-adapter-react";
+import { useEffect, useState } from "react";
+import { useSignature } from "~/components/auth/AssertAuthenticated/useSignature";
+import { useCluster } from "~/components/ClusterProvider";
+import { Redirect } from "~/components/common/Redirect";
+import { BlurBackground } from "~/components/layout/BlurBackground";
+import { Container } from "~/components/layout/Container";
+import { Flex } from "~/components/layout/Flex";
+import { Logo } from "~/components/layout/Header";
+import { Wallet } from "~/components/Wallet";
+import { useSelf } from "~/hooks/useNullableSelf";
+import { getDiscordSelf } from "~/network/discord";
+import { usePlayerStore } from "~/stores/usePlayerStore";
+import { getRoute, Routes } from "~/utils/getRoute";
+
+let pathname: string | null = "";
+
+export const View = () => {
+  const { publicKey } = useWallet();
+
+  const self = useSelf();
+  const endpoint = useCluster();
+  const linkDiscord = usePlayerStore((state) => state.linkDiscord);
+
+  const [done, setDone] = useState(false);
+
+  const signature = useSignature();
+
+  useEffect(() => {
+    const run = async () => {
+      const parsedHash = new URLSearchParams(window.location.hash.substring(1));
+      const accessToken = parsedHash.get("access_token");
+      const error = parsedHash.get("error");
+      pathname = parsedHash.get("state");
+
+      if (!accessToken || error === "access_denied") {
+        setDone(true);
+
+        return;
+      }
+
+      if (!publicKey) {
+        return;
+      }
+
+      const discordSelf = await getDiscordSelf(accessToken);
+
+      if (!discordSelf) {
+        setDone(true);
+
+        return;
+      }
+
+      linkDiscord({
+        cluster: endpoint.cluster,
+        publicKey: publicKey.toString(),
+        discordId: discordSelf.id,
+        signature,
+      });
+
+      setDone(true);
```

Comment:
This is not necessary as the linkDiscord will update the self object in the store (Also you are not `await`ing the `linkDiscord` to complete)